src/ir/typescript_types.py

```diff
@@ -216,6 +232,86 @@ def get_name(self):
         return 'undefined'
 
 
+class NumberLiteralType(TypeScriptBuiltin):
+    def __init__(self, literal, name="NumberLiteralType", primitive=False):
+        super().__init__(name, primitive)
+        self.literal = literal
+        self.supertypes.extend([ObjectType(), NumberType()])
+
+    def get_literal(self):
+        return self.literal
+
+    def is_assignable(self, other):
+        return ((isinstance(other, NumberLiteralType) and
+                  other.get_literal() == self.get_literal()) or
+                    isinstance(other, NumberType))
+
+    def get_name(self):
+        return self.name
+
+
+class StringLiteralType(TypeScriptBuiltin):
+    def __init__(self, literal, name="StringLiteralType", primitive=False):
+        super().__init__(name, primitive)
+        self.literal = literal
+        self.supertypes.extend([ObjectType(), StringType()])
+
+    def get_literal(self):
+        return '"' + self.literal + '"'
+
+    def is_assignable(self, other):
+        return ((isinstance(other, StringLiteralType) and
+                  other.get_literal() == self.get_literal()) or
+                    isinstance(other, StringType))
+
+    def get_name(self):
+        return self.name
+
+
+class LiteralTypes:
+    def __init__(self, str_limit, num_limit):
+        self.str_literals = []
+        self.num_literals = []
+        # Define max number for generated literals
+        self.str_limit = str_limit
+        self.num_limit = num_limit
+
+    def get_literal(self):
+        sl = self.gen_string_literal()
+        nl = self.gen_number_literal()
+        if ut.random.bool():
```

Comment:
I think there was a misunderstanding here.
I meant that `get_literal()` should return two types (instead of one): a string literal type, and num literal type.